bridge/docs/block_relay.md

```diff
@@ -0,0 +1,126 @@
+
+### Stage 4: replace VRF by BLS
+
+[To be completed after assessment]
+
+As aforementioned, stage 3 requires BLS signatures by Witnet nodes. Taking into account that BLS signatures are deterministic, there is a clear opportunity of using BLSs instead of VRFs for computing Proof of Eligibilities (PoEs). This is possible due to the fact that PoEs have fixed inputs for all participating nodes (e.g. data request identifier and beacon).
+
+However, before replacing VRF with BLS, it is required to perform a further analysis about the following topics and concerns:
+
+- Efficiency of BLS compared to VRF, both in Witnet nodes and Ethereum Smart Contracts (EVM).
+- Assessment of potential changes in the Witnet, both in the Witnet transaction model (e.g. commit transactions).
+- Assessment about potential impacts in the P2P network, more specifically in the `LAST_BEACON` message exchanges among peers.
+- Assess maturity of BLS (e.g. pairing functions) in terms of crypto analysis (e.g. known attacks and vulnerabilities).
+
+### Stage 5: dispute resolution
+
+A further assesment needs to be done in this matter, as to how to react in the case where the headers reported to the block relay are different than those consolidated in Witnet.
+
+## Economic Incentive: fee as low as possible
+
+Another aspect that the block relay needs to cover is the fact that nodes will have an economic incentive to report blocks, and they should not be able to set the fee as high as to make the data request non-executable. In order to cope with that we propose an inverse auction: the lowest price reported by a bridge node is the one that is set as the fee. However, we still reward all bridge nodes that participated in the correct block header consensus reporting. We apply proportionality as to how deviated the reports were with respect to the minimum. Let's work through an example:
+
+- Imagine both Alice and Bob report the same correct block header to the block relay.
+- Alice sets a report reward fee of 1 ETH, while Bob sets the fee as 9 ETH.
+- In this case, 1 ETH is established as the block report reward, of which 0.9 will go to Alice and 0.1 to Bob.
+
+With that in mind the reward that each node would obtain for someone whose reported block was successfully utilized in a PoI verification is:
+
+<p align=center>
+<img src="./images/eq_rew.png" width="300">
+</p>
+<p align=center>
+<em>Eq. 1: Reward for each of the block header reporters</em>
+</p>
+
+where |X| = x<sub>1</sub>, x<sub>2</sub>, ..., x<sub>N</sub> and each x<sub>i</sub> is the value proposed by selected node i.
+
+With this scheme all nodes are incentivized to set the fees as low as possible in favor of price competition. Ideally, all bridge nodes would set it low enough (taking into account the transaction gas and potential benefit derived from the WBI) so that they get an equal share of the rewards.
```

Comment:
I can see an scenario in which this way of sharing the rewards can assign an unbounded number of reporters a negligible amount that is way under the gas cost of performing an EVM transfer. That's damn wasteful.

I'm saying this here just for the record. When we further develop this scheme, we'll need to make sure that payouts are not triggerable by third parties. Every reporter needs to claim its own reward, yet they will likely not do so unless/until gas price goes down aggressively.